sbteclipse-core/src/main/scala/com/typesafe/sbteclipse/core/Eclipse.scala

```diff
@@ -678,6 +683,33 @@ private object Eclipse extends EclipseSDTConfig {
       fix(parts, "")
     } else filename
   }
+
+  implicit val fileEqual = new Equal[File] {
+    def equal(file1: File, file2: File): Boolean = file1 == file2
+  }
+
+  def id[A](a: A): A = a
+
+  def boolOpt(key: String): Parser[(String, Boolean)] = {
+    import sbt.complete.DefaultParsers._
+    (Space ~> key ~ ("=" ~> ("true" | "false"))) map { case (k, v) => k -> v.toBoolean }
+  }
+
+  def setting[A](key: SettingKey[A], state: State): Validation[A] =
+    key get structure(state).data match {
+      case Some(a) => a.success
+      case None => "Undefined setting '%s'!".format(key.key).failureNel
+    }
+
+  def evaluateTask[A](key: TaskKey[A], ref: ProjectRef, state: State): Validation[A] =
```

Comment:
This method is actually really bad in practice because it ignores the state returned from the task, leading to all our in-memory caches getting dropped/ignored.

We need to deprecate this and instead use Scalaz.StateT[Validation, _] I think, but let me figure that out.

ANYTIME you see someone drop a returned "state" object, it's basically a mutation "performance bug", and possibly a real bug.